### PR TITLE
Added unit tests and fixed config parser bugs

### PR DESCRIPTION
**Pull request by brwong20**

### CODE REVIEW

File: config_parser.cc [hunk 1]
```diff
@@ -198,16 +199,23 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
...
       config_stack.push(new_config);
+
+      //Counter for braces 
+      ++unmatched_braces;
+
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
-      if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
-        // Error.
+      if (last_token_type != TOKEN_TYPE_STATEMENT_END &&
+          last_token_type != TOKEN_TYPE_END_BLOCK) {
```
Comment (plantver) [inline config_parser.cc:208]: Fixed nested-statements error

File: config_parser.cc [hunk 2]
```diff
@@ -198,16 +199,23 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
...
         break;
       }
       config_stack.pop();
+      --unmatched_braces;
     } else if (token_type == TOKEN_TYPE_EOF) {
       if (last_token_type != TOKEN_TYPE_STATEMENT_END &&
-          last_token_type != TOKEN_TYPE_END_BLOCK) {
-        // Error.
+          last_token_type != TOKEN_TYPE_END_BLOCK ||
+          unmatched_braces != 0) {
```
Comment (plantver) [inline config_parser.cc:217]: Fixed unbalanced braces error.

File: config_parser_test.cc [hunk 3]
```diff
@@ -9,3 +13,53 @@ TEST(NginxConfigParserTest, SimpleConfig) {
...
+TEST(NginxConfigTest, ToString) {
+	NginxConfigStatement statement;
+	statement.tokens_.push_back("foo");
+	statement.tokens_.push_back("bar");
+	EXPECT_EQ(statement.ToString(0), "foo bar;\n");
+}
+
+class NginxStringConfigTest: public ::testing::Test {
+protected:
+	bool ParseString(const std::string& config_string){
```
Comment (plantver) [inline config_parser_test.cc:27]: Good fixture structure.

Comment (brwong20) [reply config_parser_test.cc:27]: Thanks for the review!

**Pull request merged**